### DISCUSSION TITLE
Copilot Proxy Support

### DISCUSSION THREAD

**Discussion by JustinGrote** [betalgo/openai#125] [Q&A]
The copilot proxy exposed by the Github Copilot CLI is https://copilot-proxy.githubusercontent.com/

The completion URI follows the same similar contexts and I imagine it will work fine here:
https://copilot-proxy.githubusercontent.com/v1/engines/copilot-labs-codex/completions

I'm pretty sure it just proxies an Azure ML model based on the headers, so I propose adding a new provider type that is either:
1. Copilot Specific
2. Allow to specify an arbitrary base domain that isn't Azure-based.

**Comment by kayhantolga** [comment]
you can always set your custom basedomain, library support this functionality. However, the links you sent don't work also I didn't understand quite well what you meant 🤔 .  
If you can provide me some documentation links, I can add support. 

https://github.com/betalgo/openai/blob/master/OpenAI.SDK/OpenAiOptions.cs
If you need any help with setting a custom base domain, just let me know.

**Reply by JustinGrote** [reply]
I think I misread how this works:
https://github.com/betalgo/openai/blob/d32a9a88974e17a34ba1ac58ea95a3aa5c90fa02/OpenAI.SDK/OpenAiOptions.cs#L85-L90

However when I tried to set a basedomain for copilot I got an invalid URL error, I'll see if I can debug it to find what URL it was attempting.

**Reply by kayhantolga** [reply]
Do you know Where can I find documentation for co-pilot? Also, do you have any sample basedomain for copilot?

**Reply by JustinGrote** [reply]
It's basically a private API on the backend, but it uses the same completions API as AzureML. here's an example implementation:
https://github.com/copilotplayground/CopilotDev.NET

**Reply by kayhantolga** [reply]
I don't think the current implementation would work with it. You may need to write a custom endpoint provider.
https://github.com/betalgo/openai/tree/master/OpenAI.SDK/EndpointProviders
